compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -224,30 +255,43 @@ impl<'interner> Monomorphizer<'interner> {
         main_meta.function_signature()
     }
 
+    fn compile_main_debug(
+        &mut self,
+        main_id: node_interner::FuncId,
+        field_names: &HashMap<u32, String>,
+    ) -> FunctionSignature {
+        self.debug_field_names = field_names.clone();
+        self.compile_main(main_id)
+    }
+
```

Comment:
```suggestion
```
We can remove this function and just pass `field_names` to Monomorphizer's constructor.